sdk/search/azure-search/src/test/java/com/azure/search/DocumentConverterTests.java

```diff
@@ -242,15 +243,55 @@ public void canReadComplexCollection() {
         Assert.assertEquals(expectedDoc, actualDoc);
     }
 
+    @Test
+    public void canReadArraysOfMixedTypes() {
+        // Azure Cognitive Search won't return payloads like this; This test is only for pinning purposes.
+        String json = "{\"field\": [\"hello\", 123, 3.14, { \"type\": \"Point\", \"coordinates\": [-122.131577, 47.678581] }, { \"name\": \"Arthur\", \"quest\": null }] }";
+
+        GeoPoint point = GeoPoint.create(47.678581, -122.131577);
+        Document innerDoc = new Document();
+        innerDoc.put("name", "Arthur");
+        innerDoc.put("quest", null);
+        List<Object> value = Arrays.asList("hello", 123, 3.14, point, innerDoc);
+
+        Document expectedDoc = new Document();
+        expectedDoc.put("field", value);
+
+        Document actualDoc = deserialize(json);
+        Assert.assertEquals(expectedDoc, actualDoc);
+    }
+
     @Test
     public void dateTimeStringsAreReadAsDateTime() {
         String json = "{\"field1\":\"".concat(testDateString).concat("\",\"field2\" : [\"").concat(testDateString).concat("\", \"").concat(testDateString).concat("\"]}");
-        Document expectedDoc = new Document() {
-            {
-                put("field1", testDate);
-                put("field2", Arrays.asList(testDate, testDate));
-            }
-        };
+        Document expectedDoc = new Document();
+        expectedDoc.put("field1", testDate);
+        expectedDoc.put("field2", Arrays.asList(testDate, testDate));
+
+        Document actualDoc = deserialize(json);
+        Assert.assertEquals(expectedDoc, actualDoc);
+    }
+
+    @Test
+    public void emptyArraysReadAsObjectArrays() {
+        String json = "{ \"field\": [] }";
+
+        // With no elements, we can't tell what type of collection it is, so we default to object.
+        Document expectedDoc = new Document();
+        expectedDoc.put("field", new ArrayList<>());
+
+        Document actualDoc = deserialize(json);
+        Assert.assertEquals(expectedDoc, actualDoc);
+    }
+
+    @Test
+    public void arraysWithOnlyNullsReadAsStringArrays() {
+        String json = "{ \"field\": [null, null] }";
+
+        // With only null elements, we can't tell what type of collection it is. For backward compatibility, we assume type string.
```

Comment:
I'm assuming that `Arrays.asList(null, null)` is returning an `ArrayList` of `Object`, not `String`. This is fine (since this SDK has never shipped, there are no backcompat concerns), but the comment and name of the test no longer reflect reality.

Reply:
I'll update this to make sure it uses `List<String>`.